src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs

```diff
@@ -71,13 +71,47 @@
 
         internal static string ReturnCommonPathRootFromTargetSpecifiersIfOneExists(IEnumerable<string> targetFileSpecifiers)
         {
+            var fileSpecifierDirectories = new SortedSet<string>(targetFileSpecifiers.Select(s => (Path.GetDirectoryName(Path.GetFullPath(s)) + @"\")),
+                                                                 StringComparer.OrdinalIgnoreCase);
 
-            var fileSpecifierDirectories = new HashSet<string>(targetFileSpecifiers.Select(s => Path.GetDirectoryName(Path.GetFullPath(s)) + @"\"),
-                                                               StringComparer.OrdinalIgnoreCase);
+            string smallestPath = fileSpecifierDirectories.First();
+            fileSpecifierDirectories.Remove(smallestPath);
 
-            return fileSpecifierDirectories.Count == 1
-                ? fileSpecifierDirectories.First()
-                : string.Empty;
+            for (int i = 0; i < smallestPath.Length; i++)
+            {
+                // We don't need to iterate all characters of the smallestPath if we don't have anything
+                // to compare against.
+                if (fileSpecifierDirectories.Count == 0)
+                {
+                    break;
+                }
+
+                foreach (string specifier in fileSpecifierDirectories)
+                {
+                    if (char.ToLowerInvariant(smallestPath[i]) != char.ToLowerInvariant(specifier[i]))
+                    {
+                        smallestPath = smallestPath.Substring(0, i);
+                        break;
+                    }
+                }
+            }
+
+            if (smallestPath.Length == 0)
+            {
```

Comment:
>smallestPath [](http://example.com/codeflow?start=33&length=12)

Can you just use LastIndexOf on the slash?

Reply:
yes, we can.
re-worked the code a bit to improve that.

thanks for the suggestion!